controllers/keda/util/predicate.go

```diff
@@ -61,3 +63,25 @@ func (ScaleObjectReadyConditionPredicate) Update(e event.UpdateEvent) bool {
 
 	return false
 }
+
+type PausedPredicate struct {
+	predicate.Funcs
+}
+
+func (PausedPredicate) Update(e event.UpdateEvent) bool {
+	if e.ObjectOld == nil || e.ObjectNew == nil {
+		return false
+	}
+
+	newAnnotations := e.ObjectNew.GetAnnotations()
+	oldAnnotations := e.ObjectOld.GetAnnotations()
+	if newAnnotations != nil && oldAnnotations != nil {
```

Comment:
What if the annotation was not specified in the past, but was added now. In that case, it will not be considered (unless I'm mistaken)

Reply:
This is also why I asked the e2e test change for update as this would have been caught there AFAIK